### PR TITLE
Update hyper version to fix security vulnerability

### PR DESCRIPTION
**Pull request by romoh**
**What this PR does / why we need it**:
The PR does 2 things:
- The current used version of hyper crate has a security vulnerability and fails `cargo audit`. This updates it to a version that's compatible with our code && has the issue fixed. 
- Run a cargo update since we are updating the patch version anyways.

**Special notes for your reviewer**:
N/A

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [x] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [x] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [x] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

**Comment by bfjelds** [comment]
It looks like the h2 dependency changed.  Is it a problem that we aren't using the h2 fork?

**Comment by romoh** [comment]
> It looks like the h2 dependency changed. Is it a problem that we aren't using the h2 fork?

h2 dependency is still there in the Cargo.lock and the version is locked in the toml file, so I highly doubt it.
The workflow failed in 'installing kubernetes'.. I'm rerunning the jobs to see if that was just a hiccup.

**Comment by romoh** [comment]
> > It looks like the h2 dependency changed. Is it a problem that we aren't using the h2 fork?
> 
> h2 dependency is still there in the Cargo.lock and the version is locked in the toml file, so I highly doubt it.
> The workflow failed in 'installing kubernetes'.. I'm rerunning the jobs to see if that was just a hiccup.

Error below.. @bfjelds  any clue what might be going on here?
  Preparing to unpack .../8-kubeadm_1.16.15-00_amd64.deb ...
  Unpacking kubeadm (1.16.15-00) ...
  **Errors were encountered while processing:
   /tmp/apt-dpkg-install-d8lyhl/2-containers-common_100%3a1-7_all.deb
  E: Sub-process /usr/bin/dpkg returned an error code (1)
  Error: Process completed with exit code 100.**

**Comment by bfjelds** [comment]
I dont know why this started happening, but the solution is here: https://github.com/deislabs/akri/pull/244/commits/2e168fdc641f5b7182491dc649debcafc202e7ee

**Review by bfjelds — APPROVED** [review]

File: .github/workflows/run-test-cases.yml [hunk 1]
```diff
@@ -186,7 +186,7 @@ jobs:
       name: Install Kubernetes
       run: |
         sudo apt-get update -y
-        sudo apt-get install -y --allow-downgrades kubelet=${{ matrix.kube-version }} kubeadm=${{ matrix.kube-version }} kubectl=${{ matrix.kube-version }} 
+        sudo apt-get install -o Dpkg::Options::="--force-overwrite" -y --allow-downgrades kubelet=${{ matrix.kube.version }} kubeadm=${{ matrix.kube.version }} kubectl=${{ matrix.kube.version }}
```
Comment (kate-goldenring) [inline .github/workflows/run-test-cases.yml:189]: i think if `kube.version` is changed back to `kube-version` this should work. Daz is also changing the tst matrix in this commit https://github.com/deislabs/akri/commit/2e168fdc641f5b7182491dc649debcafc202e7ee, which is why the `kube-version` variable was changed

**Review by kate-goldenring — CHANGES_REQUESTED** [review]

File: .github/workflows/run-test-cases.yml (hunk 1 above)
Comment (romoh) [reply .github/workflows/run-test-cases.yml:189]: I tried it locally on a vm and it still fails with the same error in the workflow. Looking around to understand the error.

Comment (romoh) [reply .github/workflows/run-test-cases.yml]: or maybe not.. I was missing the last 00 in the version when I tried it locally. Hopefully it works now.

**Review by kate-goldenring — APPROVED** [review]

**Pull request merged**